src/test/java/com/salesforce/dynamodbv2/PutTest.java

```diff
@@ -74,12 +76,12 @@ void putOverwrite(TestArgument testArgument) {
             assertThat(putItemRequest.getItem(), is(new HashMap<>(itemToOverwrite))); // assert no side effects
             assertEquals(TABLE1, putItemRequest.getTableName()); // assert no side effects
             assertThat(getItem(testArgument.getHashKeyAttrType(), testArgument.getAmazonDynamoDb(), TABLE1),
-                       is(itemToOverwrite));
+                is(itemToOverwrite));
         });
     }
 
-    @TestTemplate
-    @ExtendWith(TestTemplateWithDataSetup.class)
+    @ParameterizedTest(name = "{arguments}")
+    @ArgumentsSource(DefaultArgumentProvider.class)
     void putHkRkTable(TestArgument testArgument) {
         testArgument.getOrgs().forEach(org -> {
             MT_CONTEXT.setContext(org);
```

Comment:
Maybe worth considering an `MT_CONTEXT.withContext(org, () -> ...)` that sets the context to the given org and upon completion sets the context back to the previous value (which is likely null in most cases). To collapse these two lines, there could also be a `testArgument.getOrgs().withEachContext(() -> ...)` that iterates and calls that method for each loop.

Reply:
Will do.